Modified startup script and server

bheesham opened the pull request:
- The server's configuration can now be set in `etc/server_{prod, devel}.yml`.
- The installation script installs redis.

Commit:
Now copies the client stuff to another folder.

	modified:   bin/install_server.sh

Commit:
New start script.

	renamed:    bin/start_dev.sh -> bin/start.sh

Commit:
No hard coded timeout and port for server port.

	new file:   bin/.start.sh.swp
	modified:   bin/start.sh
	new file:   etc/server_devel.yml
	new file:   etc/server_prod.yml
	modified:   src/server/index.js
	modified:   src/server/package.json

Commit:
Fixed a typo(?)

	modified:   README.md

Commit:
Installs redis

	deleted:    bin/.start.sh.swp
	modified:   bin/install_server.sh
	modified:   bin/start.sh

Commit:
Merge branch 'master' of github.com:CarletonU-COMP2406-W2013/Steamed-Pears

psimonyi commented on etc/server_devel.yml:
```diff
@@ -0,0 +1,4 @@
+server_port: 25414
+req_timeout: 3000
+
+# vim: ft=yaml
```
In modelines, I kind of prefer the full versions of commands (i.e. `filetype=yaml`), though I'd compromise to get this all on one line (especially since `ft` is a common enough option; more esoteric options really need to be readable).
It's probably a good idea to grab the standard vim modeline from the wiki, too.

psimonyi commented on src/server/index.js:
```diff
@@ -7,14 +7,18 @@
-var serverPort = 20193;
-var requestTimeout = 3000; // ms
+
+var server_config = require(process.env.SERVER_CONF);
```
Woah! You just _execute_ the YAML file?! Isn't there something like YAML.parse?

psimonyi commented on src/server/package.json:
```diff
@@ -28,7 +28,8 @@
     "cors":             ">=0.0.5",
-    "proxy-middleware": ">=0.3.1"
+    "proxy-middleware": ">=0.3.1",
+		"js-yaml": 					">=2.0.3"
```
Spacing

psimonyi commented on bin/install_server.sh:
```diff
@@ -14,6 +14,9 @@ ROOT_DIR=`pwd`
 # Configuration
 ######################################################################
+
+CLIENT_TARGET_DIR=~/webapps/cr_front
```
I don't like the idea of deploying to a home directory -- especially a hard-coded path.

spratt replied:
Is there an obvious better way to do this?

spratt replied:
On this line, we install to a home directory.

spratt replied:
Let's talk about this on this line.

psimonyi replied:
I think the Right Thing To Do is have a good default like `/srv/code_review` or maybe `/var/www/code_review` and check an environment variable or command-line parameter for a user-specified value. (Think of installing programs, where the default is usually `/usr/bin` but you can do `./configure --prefix=$HOME/local` instead.) If you go for environment variables, look up Parameter Expansion in `bash(1)`; if you go for command-line parameters, look up getopt.

psimonyi replied:
@bheesham says we can be sure of write access to the home directory.
I think it is better to have a default for when it's your own server, and make it easy to override the default. (And you were worrying about what happens when installing as root before...)

bheesham replied:
Hmm, yeah.
I guess when you run it as root, you'll probably want to put nginx, npm, and redis-server in `/usr/bin``.

As for not having the permissions to put it there, what would you suggest we do? Because changing PATH seems weird, and I feel gross. Because in development, I'd be running this a lot of times.

psimonyi commented on bin/install_server.sh:
```diff
@@ -22,16 +25,42 @@ NODE_VERSION=0.10.2
+make
+
+echo Copying client files into the specified directory 
+cp -R $ROOT_DIR/src/client/* $CLIENT_TARGET_DIR/
```
It seems strange to be dealing with client files in `install_server.sh`. The file description at the top makes no mention of this.

psimonyi replied:
The main problem is not mentioning this in the description of what this file does, as @spratt says. A minor problem is that the file name `install_server.sh` suggests it's installing the server component, when it really seems to be installing everything.

bheesham replied:
So... do we not want to install everything?

psimonyi commented on bin/start.sh:
```diff
@@ -18,7 +18,7 @@ NODE_BIN=$ROOT_DIR/bin/exe/node/bin
 NODE=$NODE_BIN/node
 
-DB=redis-server
+DB=$ROOT_DIR/bin/exe/redis/src/redis-server
```
`bin/exe/redis/src` what?

spratt committed:
Merged persona server

psimonyi commented on bin/start.sh:
```diff
@@ -62,7 +65,24 @@ ln -s -f $DB_LOG.$TIME.log $DB_LINK
+
+
+
+
```
Extra blank lines at EOF

psimonyi commented:
It seems this still installs node, npm and redis even in development mode.

cdelahousse commented on bin/install_server.sh:
```diff
@@ -22,16 +25,42 @@ NODE_VERSION=0.10.2
+echo Installing component: redis
+
+echo - downloading...
+wget -q http://redis.googlecode.com/files/$REDIS_F.tar.gz
```
Shouldn't we host a reliable version as we've done with our front end dependencies?

psimonyi replied:
I thought we were hosting things only when we couldn't find a good download link.

spratt replied:
I only hosted things when the only option was over https or there was no real download link.

spratt committed:
Now parsing YAML safely

spratt committed:
Merge pull request #2 from spratt/yaml

Safely parsing YAML

bheesham commented:
@psimonyi , If you already have the things install, why install them again?
You should be using `start.sh`

``` bash
alias vim=sudo yum install vim && vim
```

spratt commented:
Outstanding issues for @bheesham:
- spacing in  `src/server/package.json`
- installing to a home directory in `bin/install_server.sh`
- add copying of client files to description of `bin/install_server.sh`

bheesham commented:
[Currently it does copy client files.](https://github.com/bheesham/Steamed-Pears/blob/master/bin/install_server.sh#L66)

spratt commented:
The problem is that there is no mention of copying client files in the description in the header of `bin/install_server.sh`

bheesham commented:
Ah, okay.

Also, for installation of the server to a home directory.
When you install this as root, and run it as another person, would that user not be able to find it? Because root's home dir is different.

spratt commented:
Why not reply directly to the comments on the lines of code?  It's easier to understand when there's context to what you're saying.

bheesham commented:
Because I'm replying directly to your comment.

spratt commented:
Okay, in general I agree with you.  So why are we installing to a home directory?

psimonyi commented:
This will also fix #123.

bheesham commented:
Because the user we're installing it as, for sure has the permissions to write files there.
On WebFaction (our production server), that's the only place we're allowed to write.

psimonyi commented:
Let's discuss the home directory thing on its line.

Commit:
Now installs to $PREFIX/bleh, doesn't handle copying client stuff.

	new file:   bin/.install_server.sh.swp
	deleted:    bin/exe/.folder
	modified:   bin/install_server.sh
	modified:   etc/server_devel.yml
	modified:   etc/server_prod.yml

Commit:
Now installs to $PREFIX/usr/bin and uses $PREFIX/tmp

	modified:   .gitignore
	modified:   bin/install_server.sh

Commit:
Merge branch 'master' of github.com:bheesham/Steamed-Pears

Conflicts:
	etc/server_devel.yml
	etc/server_prod.yml

	modified:   .gitignore
	new file:   .gitmodules
	modified:   INSTALL.md
	modified:   README.md
	modified:   bin/helpers.sh
	modified:   bin/start.sh
	deleted:    docs/.folder
	modified:   etc/server_devel.yml
	modified:   etc/server_prod.yml
	new file:   http_api.md
	modified:   src/client/index.html
	new file:   src/client/scripts/login.js
	modified:   src/client/scripts/main.js
	modified:   src/client/scripts/view.js
	new file:   src/client/styles/persona-buttons.css
	modified:   src/client/styles/view.css
	modified:   src/server/dev_mode.js
	modified:   src/server/index.js
	modified:   src/server/requestHandlers.js
M

Commit:
Removed some tabs :'(

	modified:   src/server/package.json

psimonyi commented on bin/start.sh:
```diff
@@ -1,23 +1,35 @@
+if $PROD; then
+  NPM=$ROOT_DIR/bin/exe/node/bin/npm
+  NODE=$ROOT_DIR/bin/exe/node/bin/node
+  DB=$ROOT_DIR/bin/exe/redis/src/redis-server
```
These paths are no longer correct, right?

psimonyi commented on .gitignore:
```diff
@@ -1,3 +1,5 @@
+*.swp
```
This should probably be `.*.swp` (dotfiles don't normally match `*` in globs). You've committed a swapfile in spite of this, anyway.

spratt replied:
Ugh.  Let's get rid of the swapfile.

psimonyi commented on bin/install_server.sh:
```diff
@@ -22,16 +22,61 @@ NODE_VERSION=0.10.2
+
+echo - installing redis...
+cp $TMP/$REDIS_F/src/redis-server $BIN/redis-server
+cp $TMP/$REDIS_F/src/redis-cli $BIN/redis-cli
```
It would probably be less fragile to do `make PREFIX=$PREFIX install`.

bheesham replied:
Added.

psimonyi commented on bin/install_server.sh:
```diff
@@ -22,16 +22,61 @@ NODE_VERSION=0.10.2
+BIN=$PREFIX/usr/bin
+
+mkdir -p $TMP 2> /dev/null
+mkdir -p $BIN 2> /dev/null
```
We probably want to know about any errors here. `mkdir -p` is silent when the directory exists, at least for me.

bheesham replied:
Added.

psimonyi commented on bin/install_server.sh:
```diff
@@ -22,16 +22,61 @@ NODE_VERSION=0.10.2
+
 ######################################################################
 
+TMP=$PREFIX/tmp
```
`TMP=${TEMP:-${TMP:-/tmp}}` would use the first non-empty path of `$TEMP`, `$TMP`, and `/tmp`. Any of those is probably a good temp directory. In any case, we should probably clean up after ourselves...

bheesham replied:
Should have noted that I removed this after.

psimonyi commented on bin/install_server.sh:
```diff
@@ -22,16 +22,61 @@ NODE_VERSION=0.10.2
 ######################################################################
 
+TMP=$PREFIX/tmp
+BIN=$PREFIX/usr/bin
```
Typically, `PREFIX` values are things like `/usr/local` or `$HOME` or `/usr`; in all cases, I'd expect binaries to go in `$PREFIX/bin`.

bheesham replied:
Well, if it **is** /usr/, that sounds weird. /usr/node ? No thanks?

On 09/Apr/13 12:47 AM, Peter Simonyi wrote:

> In bin/install_server.sh:
> 
> >  ######################################################################
> > 
> > +TMP=$PREFIX/tmp
> > +BIN=$PREFIX/usr/bin
> 
> Typically, |PREFIX| values are things like |/usr/local| or |$HOME| or 
> |/usr|; in all cases, I'd expect binaries to go in |$PREFIX/bin|.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/CarletonU-COMP2406-W2013/Steamed-Pears/pull/126/files#r3707631.

psimonyi replied:
I'd still kind of like to see a default value for PREFIX, though I won't insist on it. However, the docs should say somewhere that you can use it (and generally tell you how to run this thing).

psimonyi replied:
If `PREFIX` is `/usr`, then `BIN=$PREFIX/bin` would make `/usr/bin/node`, no?

bheesham replied:
Then we create /usr/tmp? Crazy talk.

psimonyi replied:
No, you put the temp files in `$TEMP` (or `$TMP`, or `/tmp` if neither is set), and install binaries to `$PREFIX/bin`.

bheesham replied:
I can't compile in the `/tmp` folder. So, there is no installing binaries without compilation.

psimonyi commented:
Blocking on the paths in `start.sh` and the swapfile; the other stuff can be done/discussed later if you want.

psimonyi commented:
You should also delete `bin/exe/.folder`.

psimonyi commented:
Perhaps `stop_dev.sh` should be renamed too. Does it work in production mode?

Commit:
Removed the swap file entry from .gitignore

	modified:   .gitignore

Commit:
Now mostly non-blocking.

  * Fixed the paths in ```bin/start.sh```
  * Fixed the paths in ```bin/install_server.sh```
  * Now installs redis (make PREFIX=$PREFIX install)
  * Removed the o-displaying-stderr
  * Renamed ```bin/stop_dev.sh``` to ```bin/stop.sh```

	deleted:    bin/.install_server.sh.swp
	modified:   bin/install_server.sh
	modified:   bin/start.sh
	renamed:    bin/stop_dev.sh -> bin/stop.sh

bheesham commented:
It should now be mostly non-blocking. I didn't add `TMP=${...`, because I don't have permissions to write to compile in`/tmp`.

**THIS WORKS FOR ME IN PRODUCTION ON WEBFACTION.**

Commit:
Changed references to different files. Updated the docs.

	modified:   INSTALL.md
	modified:   bin/start.sh

Commit:
Merge branch 'master' of github.com:CarletonU-COMP2406-W2013/Steamed-Pears

Conflicts:
	INSTALL.md

	modified:   .gitignore
	modified:   INSTALL.md
	new file:   docs
	deleted:    src/bookmarklet/.gitignore
	modified:   src/client/.gitignore
	new file:   src/client/images/simple_pear.png
	modified:   src/client/index.html
	deleted:    var/.folder
	modified:   var/.gitignore
	deleted:    var/db/.folder
	new file:   var/db/.gitignore
	deleted:    var/logs/.folder
	new file:   var/logs/.gitignore

Commit:
Added the temp dir.

	modified:   bin/install_server.sh

Commit:
Merge branch 'master' of github.com:CarletonU-COMP2406-W2013/Steamed-Pears

Conflicts:
	src/server/README.md

	modified:   .gitignore
	modified:   INSTALL.md
	modified:   README.md
	modified:   docs
	modified:   http_api.md
	deleted:    src/client/INSTALL
	modified:   src/client/index.html
	new file:   src/client/scripts/anticsrf.js
	modified:   src/client/scripts/main.js
	modified:   src/client/scripts/view.js
	deleted:    src/server/.gitignore
	deleted:    src/server/INSTALL
	deleted:    src/server/README.md
	modified:   src/server/index.js
	modified:   src/server/package.json
	modified:   src/server/requestHandlers.js

spratt commented:
Looks good.  I'm going to pull this down and try it out on my webfaction.

spratt commented:
Fantastic.  git gives errors pushing to webfaction if I don't have the docs submodule pulled in, but after installing it, it won't upload the docs to webfaction....fine, whatever git.  You suck.

bheesham commented:
Huh... I never got those errors.

On 12/Apr/13 6:30 PM, Simon David Pratt wrote:

> Fantastic. git gives errors pushing to webfaction if I don't have the 
> docs submodule pulled in, but after installing it, it won't upload the 
> docs to webfaction....fine, whatever git. You suck.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/CarletonU-COMP2406-W2013/Steamed-Pears/pull/126#issuecomment-16321021.

spratt commented:
Redis fails to build:

```
make[3]: Entering directory `/home/sdp/tmp/redis-2.6.12/deps/jemalloc'
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/jemalloc.o src/jemalloc.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/arena.o src/arena.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/atomic.o src/atomic.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/base.o src/base.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/bitmap.o src/bitmap.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/chunk.o src/chunk.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/chunk_dss.o src/chunk_dss.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/chunk_mmap.o src/chunk_mmap.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/ckh.o src/ckh.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/ctl.o src/ctl.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/extent.o src/extent.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/hash.o src/hash.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/huge.o src/huge.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/mb.o src/mb.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/mutex.o src/mutex.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/prof.o src/prof.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/quarantine.o src/quarantine.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/rtree.o src/rtree.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/stats.o src/stats.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/tcache.o src/tcache.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/util.o src/util.c
gcc -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops  -c -D_GNU_SOURCE -D_REENTRANT -Iinclude -Iinclude -o src/tsd.o src/tsd.c
ar crus lib/libjemalloc.a src/jemalloc.o src/arena.o src/atomic.o src/base.o src/bitmap.o src/chunk.o src/chunk_dss.o src/chunk_mmap.o src/ckh.o src/ctl.o src/extent.o src/hash.o src/huge.o src/mb.o src/mutex.o src/prof.o src/quarantine.o src/rtree.o src/stats.o src/tcache.o src/util.o src/tsd.o
make[3]: Leaving directory `/home/sdp/tmp/redis-2.6.12/deps/jemalloc'
make[2]: Leaving directory `/home/sdp/tmp/redis-2.6.12/deps'
    CC adlist.o
    CC ae.o
    CC anet.o
    CC dict.o
    CC redis.o
    CC sds.o
    CC zmalloc.o
    CC lzf_c.o
    CC lzf_d.o
    CC pqsort.o
    CC zipmap.o
    CC sha1.o
    CC ziplist.o
    CC release.o
    CC networking.o
    CC util.o
    CC object.o
    CC db.o
    CC replication.o
    CC rdb.o
    CC t_string.o
    CC t_list.o
    CC t_set.o
    CC t_zset.o
    CC t_hash.o
    CC config.o
    CC aof.o
    CC pubsub.o
    CC multi.o
    CC debug.o
    CC sort.o
    CC intset.o
    CC syncio.o
    CC migrate.o
    CC endianconv.o
    CC slowlog.o
    CC scripting.o
    CC bio.o
    CC rio.o
    CC rand.o
    CC memtest.o
    CC crc64.o
    CC bitops.o
    CC sentinel.o
    LINK redis-server
zmalloc.o: In function `zmalloc_used_memory':
/home/sdp/tmp/redis-2.6.12/src/zmalloc.c:223: undefined reference to `__sync_add_and_fetch_4'
collect2: ld returned 1 exit status
make[1]: *** [redis-server] Error 1
make[1]: Leaving directory `/home/sdp/tmp/redis-2.6.12/src'
make: *** [all] Error 2
cd src && make install
make[1]: Entering directory `/home/sdp/tmp/redis-2.6.12/src'
    LINK redis-server
zmalloc.o: In function `zmalloc_used_memory':
/home/sdp/tmp/redis-2.6.12/src/zmalloc.c:223: undefined reference to `__sync_add_and_fetch_4'
collect2: ld returned 1 exit status
make[1]: *** [redis-server] Error 1
make[1]: Leaving directory `/home/sdp/tmp/redis-2.6.12/src'
make: *** [install] Error 2
- installing redis...
cp: cannot stat `/home/sdp/tmp/redis-2.6.12/src/redis-server': No such file or directory
cp: cannot stat `/home/sdp/tmp/redis-2.6.12/src/redis-cli': No such file or directory
```

spratt commented on bin/install_server.sh:
```diff
@@ -22,16 +22,69 @@ NODE_VERSION=0.10.2
+
+echo - compiling redis...
+make
+make PREFIX=$PREFIX install
```
I think this should be `PREFIX=$PREFIX make install`

spratt commented:
The install script works fine on my mac using clang 4.2, but it won't compile on my webfaction host using gcc 4.1.  I have a suspicion that gcc 4.1 is a little out of date.

spratt committed:
Fixed webfaction bug

spratt commented:
Bhee, I fixed the issues I mentioned above in a pull request to your master branch.

spratt commented:
[blocking] I can't execute the node binary file that we download using this script.

spratt commented:
For the record, I found the solution to the redis compile bug from: http://www.eschrade.com/page/undefined-reference-to-__sync_add_and_fetch_4/

spratt commented:
`uname -m` returns an architecture that my mac's version of gcc doesn't support...and the redis build uses gcc in particular at one stage of its build (as opposed to $CC which would use clang).

spratt committed:
TESTING won't build on mac

spratt committed:
TESTING fix for #153 and #152

spratt committed:
Fixed webfaction deployment issues

fixes #153
fixes #152

	modified:   bin/build_production_all.sh

spratt committed:
Added documentation for build_production_all.sh

closes #151

	modified:   INSTALL.md

spratt committed:
No longer creating a new shell in build

	modified:   bin/build_production_all.sh

spratt committed:
Fixed paths, no longer using $PREFIX/usr/bin

	modified:   bin/start.sh

spratt committed:
Now copying images in build script

	modified:   bin/build_production_all.sh

spratt committed:
Fixed build

fixes # 154

	modified:   main.js
	modified:   view.js

spratt committed:
Now properly removing old build

spratt committed:
Merge pull request #3 from spratt/prod_deploy

Fixed webfaction bug

spratt commented:
This now works on the main site.  See: http://review.steamedpears.com

spratt committed:
Comments now appear old->new

fixes #141

	modified:   src/server/requestHandlers.js

spratt committed:
Now dealing nicely with 404s

	modified:   src/client/scripts/view.js

spratt committed:
Merge pull request #4 from spratt/prod_deploy

Prod deploy

spratt committed:
Fixed tabs

spratt committed:
Merge pull request #5 from spratt/prod_deploy

Fixed tabs

spratt commented:
@psimonyi you should take a look at this request now.  I think a lot of issues have been fixed.

psimonyi commented on bin/install_server.sh:
```diff
@@ -14,24 +15,80 @@ ROOT_DIR=`pwd`
 
 # A valid node version
-NODE_VERSION=0.10.2
+NODE_VERSION=0.10.4
```
The "Requirements" section should probably list this version of Node.

spratt replied:
Fixed.

psimonyi commented on bin/install_server.sh:
```diff
@@ -14,24 +15,80 @@ ROOT_DIR=`pwd`
 NODE_ARCH=64
 
+# Redis version
+REDIS_VERSION=2.6.12
```
Same here. The docs say we've tested with 2.6.10.

spratt replied:
Fixed.

psimonyi commented on bin/install_server.sh:
```diff
@@ -14,24 +15,80 @@ ROOT_DIR=`pwd`
+cd $TMP/$REDIS_F 
+
+echo - compiling redis...
+CFLAGS=-march=`uname -m` make
```
Waiting on this thing.

spratt replied:
By which, Peter means that although it fixes an issue on the server it breaks the build everywhere else.

psimonyi replied:
Right. I think the appropriate thing to do is just call `make` here, and do `export CFLAGS=i686` before running the script, because this seems like a bug caused by an old version of gcc. Other possibilities, since Webfaction is currently our primary deployment target, are: have a script `webfaction_hack.bash` that does that, or try to detect the version of GCC (but that's not exactly easy).

spratt replied:
I like the export CFLAGS fix.  We can document the issue and the resolution.

psimonyi commented on bin/build_production_all.sh:
```diff
@@ -13,14 +13,12 @@ CLIENT_SCRIPTS_DIR="$CLIENT_DIR/scripts"
 #Temporary folder for the build
-TARGET_TMP="$TARGET""_tmp"
+TMP=${TEMP:-${TMP:-/tmp}}
+TARGET_TMP=$TMP/cr
```
It would be better to use mktemp for this (`mktemp -d --tmpdir="$TMP" codereview-XXXX`). In fact, just `--tmpdir` (no argument) would use `$TMPDIR` if set, and default to `/tmp`.
This can wait for later. [non-blocking]

psimonyi committed:
Made compiling Redis sane for most systems.

psimonyi committed:
Use mktemp to make the temp directory.

psimonyi committed:
Update docs submodule pointer

Commit:
Merge pull request #6 from psimonyi/bhee

Fix compiling redis; use mktemp

spratt commented:
This pull request can't be automatically merged...

Commit:
Merge branch 'master' of github.com:CarletonU-COMP2406-W2013/Steamed-Pears

bheesham commented:
It can now.

spratt commented:
Builds nicely on my mac.  Fails on the server without -march=i686.  Now testing with -march=i686.

spratt commented:
Builds successfully on server.  I'll do one last review then merge.

spratt committed:
Update INSTALL.md

Updated requirements and tested versions
